governance/checkpoint.go

```diff
@@ -31,31 +37,73 @@ func (e *Engine) Checkpoint() ([]byte, error) {
 	if len(e.enactedProposals) == 0 {
 		return nil, nil
 	}
-	snap := &checkpointpb.Proposals{
+	cp := &checkpointpb.Proposals{
 		Proposals: e.getCheckpointProposals(),
 	}
-	return proto.Marshal(snap)
+	return proto.Marshal(cp)
 }
 
 func (e *Engine) Load(ctx context.Context, data []byte) error {
-	snap := &checkpointpb.Proposals{}
-	if err := proto.Unmarshal(data, snap); err != nil {
+	cp := &checkpointpb.Proposals{}
+	if err := proto.Unmarshal(data, cp); err != nil {
 		return err
 	}
 
-	e.activeProposals = make([]*proposal, 0, len(snap.Proposals))
-	evts := make([]events.Event, 0, len(snap.Proposals))
-	for _, p := range snap.Proposals {
-		if p.Terms.ClosingTimestamp < e.timeService.GetTimeNow().Unix() {
-			// the proposal in question has expired, ignore it
-			continue
-		}
+	evts := make([]events.Event, 0, len(cp.Proposals))
+	now := e.timeService.GetTimeNow()
+	minEnact, err := e.netp.GetDuration(netparams.GovernanceProposalMarketMinEnact)
+	if err != nil {
+		e.log.Panic("failed to get proposal market min enactment duration from network parameter")
+	}
+	minAuctionDuration, err := e.netp.GetDuration(netparams.MarketAuctionMinimumDuration)
+	if err != nil {
+		e.log.Panic("failed to get proposal market min auction duration from network parameter")
+	}
+	duration := minEnact
+	// we have to choose the max between minEnact and minAuctionDuration otherwise we won't be able to submit the market successfully
+	if int64(minEnact) < int64(minAuctionDuration) {
+		duration = minAuctionDuration
+	}
+
+	for _, p := range cp.Proposals {
 		prop, err := types.ProposalFromProto(p)
 		if err != nil {
 			return err
 		}
+
+		switch prop.Terms.Change.GetTermType() {
+		case types.ProposalTermsTypeNewMarket:
+			// if the proposal is for a new market we want to restore it such that it will be in opening auction
+			if p.Terms.EnactmentTimestamp <= now.Unix() {
+				prop.Terms.EnactmentTimestamp = now.Add(duration).Unix()
+			}
+			if toSubmit, err := e.intoToSubmit(ctx, prop); err == nil {
```

Comment:
If `intoToSubmit` fails we're just silently ignoring it, is that intentional?

Reply:
I think it pretty much can't really fail (famous last words) but I'll add panic.